torch/csrc/autograd/variable.cpp

```diff
@@ -164,6 +164,12 @@ void Variable::detach_() {
   set_gradient_edge(Edge());
 }
 
+Variable Variable::as_param() const {
+  auto res = make_variable(data(), /*requires_grad=*/true);
+  res.set_version_counter(version_counter());
```

Comment:
I don't think what we're doing here is safe. If you do `v.as_param().mul_(2)`, then this won't get registered in the original graph, and the derivative will be incorrect!

Reply:
Couldn't we make it a view that requires grad?

Reply:
Why wouldn't this be tracked? This is the exact `y = x.detach(); y.required_grad = True` syntax. It's sharing the same version counter with the original tensor, so any inplace operation that breaks backward should be detected.

Reply:
It is also safer than `Variable(x.data, requires_grad=True)`.

Reply:
Oh I misread your comment. I'm sorry. Yeah, the concern is valid.